source/Ship.cpp

```diff
@@ -3585,6 +3617,111 @@ void Ship::SetTargetSystem(const System *system)
 
 
 
+// Persistent targets for mission NPCs.
+const bool Ship::HasTravelDirective() const
+{
+	return !stopovers.empty() || destinationPlanet || destinationSystem;
+}
+
+
+
+const map<const Planet *, bool> Ship::GetStopovers() const
+{
+	return stopovers;
+}
+
+
+
+const bool Ship::AllStopoversVisited() const
+{
+	bool allVisited = true;
+	if(stopovers.empty())
+		return allVisited;
+
+	for(auto const &it : stopovers)
+		allVisited &= it.second;
+	return allVisited;
+}
+
+
+
+const Planet *Ship::GetDestinationPlanet() const
+{
+	return destinationPlanet;
+}
+
+
+
+const System *Ship::GetDestinationSystem() const
+{
+	return destinationSystem;
+}
+
+
+
+// Persistent targets for mission NPCs.
+void Ship::SetDestination(const Planet *destination)
+{
+	this->destinationPlanet = destination;
```

Comment:
`this` is unnecessary here
```suggestion
	destinationPlanet = destination;
```